lib/scout_apm/auto_instrument/instruction_sequence.rb

```diff
@@ -0,0 +1,30 @@
+
+require 'scout_apm/auto_instrument/rails'
+
+module ScoutApm
+  module AutoInstrument
+    module InstructionSequence
+      
+      def load_iseq(path)
+        if Rails.controller_path?(path)
+          begin
+            new_code = Rails.rewrite(path)
+            return self.compile(new_code, File.basename(path), path)
+          rescue SyntaxError => error
+            warn "Failed to apply auto-instrumentation to #{path}: #{error}"
+           end
+        end
+        
+        return self.compile_file(path)
+      end
+    end
+    
+    # This should work (https://bugs.ruby-lang.org/issues/15572), but it doesn't.
+    # RubyVM::InstructionSequence.extend(InstructionSequence)
+
+    # So we do this instead:
+    class << ::RubyVM::InstructionSequence
```

Comment:
@ioquatix Can you put conditionals around this to prevent trying to prepend on older ruby versions? I think 2.3.1+ is the minimum.